src/Compilers/Core/Portable/StrongName/StrongNameProvider.cs

```diff
@@ -13,14 +26,43 @@ protected StrongNameProvider()
         {
         }
 
-        public abstract override bool Equals(object other);
         public abstract override int GetHashCode();
+        public override abstract bool Equals(object other);
+
+        internal abstract SigningCapability Capability { get; }
 
         /// <exception cref="IOException"></exception>
         internal abstract Stream CreateInputStream();
 
         internal abstract StrongNameKeys CreateKeys(string keyFilePath, string keyContainerName, CommonMessageProvider messageProvider);
 
-        internal abstract void SignAssembly(StrongNameKeys keys, Stream inputStream, Stream outputStream);
+        internal StrongNameKeys CommonParseKeys(StrongNameFileSystem fileSystem, string keyFilePath, ImmutableArray<string> keyFileSearchPaths, CommonMessageProvider messageProvider) {
+            try
+            {
+                string resolvedKeyFile = fileSystem.ResolveStrongNameKeyFile(keyFilePath, keyFileSearchPaths);
+                if (resolvedKeyFile == null)
+                {
+                    var message = $"{keyFilePath}: {CodeAnalysisResources.FileNotFound}";
+                    return new StrongNameKeys(StrongNameKeys.GetKeyFileError(messageProvider, keyFilePath, message));
```

Comment:
The message format should probably be a resource. Again, consider simply `throw new FileNotFound();` as `DesktopStrongNameProvider` was doing previously.

Reply:
Haha, beat you to it on this one.  I've changed it to just be `CodeAnalysisResources.FileNotFound` because the GetKeyFileError handles the file name just fine.

@jaredpar explicitly forbade throwing from throwing immediately inside a `try`.